waspc/cli/src/Wasp/Cli/Command/CreateNewProject/ProjectDescription.hs

```diff
@@ -63,46 +64,50 @@ instance Show NewProjectAppName where
     - Template name is required, we ask the user to choose from available templates.
 -}
 obtainNewProjectDescription :: NewProjectArgs -> [StarterTemplate] -> Command NewProjectDescription
-obtainNewProjectDescription NewProjectArgs {_projectName = projectNameArg, _templateName = templateNameArg} starterTemplates =
-  case projectNameArg of
-    Just projectName -> obtainNewProjectDescriptionFromCliArgs projectName templateNameArg starterTemplates
-    Nothing -> obtainNewProjectDescriptionInteractively templateNameArg starterTemplates
-
-obtainNewProjectDescriptionFromCliArgs :: String -> Maybe String -> [StarterTemplate] -> Command NewProjectDescription
-obtainNewProjectDescriptionFromCliArgs projectName templateNameArg availableTemplates =
-  obtainNewProjectDescriptionFromProjectNameAndTemplateArg
-    projectName
-    templateNameArg
-    availableTemplates
-    (return defaultStarterTemplate)
-
-obtainNewProjectDescriptionInteractively :: Maybe String -> [StarterTemplate] -> Command NewProjectDescription
-obtainNewProjectDescriptionInteractively templateNameArg availableTemplates = do
-  projectName <- liftIO $ Interactive.askForRequiredInput "Enter the project name (e.g. my-project)"
-  obtainNewProjectDescriptionFromProjectNameAndTemplateArg
-    projectName
-    templateNameArg
-    availableTemplates
-    (liftIO askForTemplateName)
-  where
-    askForTemplateName = Interactive.askToChoose "Choose a starter template" $ fromList availableTemplates
-
-obtainNewProjectDescriptionFromProjectNameAndTemplateArg ::
-  String ->
-  Maybe String ->
-  [StarterTemplate] ->
-  Command StarterTemplate ->
-  Command NewProjectDescription
-obtainNewProjectDescriptionFromProjectNameAndTemplateArg projectName templateNameArg availableTemplates obtainTemplateWhenNoArg = do
+obtainNewProjectDescription NewProjectArgs {_projectName = projectNameArg, _templateName = templateNameArg} starterTemplates = do
+  projectName <- maybe askForName return projectNameArg
+  appName <-
+    either throwProjectCreationError pure $
+      parseWaspProjectNameIntoAppName projectName
+
+  let prefersInteractive = isNothing projectNameArg
```

Comment:
This way of detecting the preference was implicit in the previous version of the code, we now make it explicit